### PR TITLE
net/miniupnpd: update to 2.3.1

### PR DESCRIPTION
**Pull request by Tawmu**
Updates miniupnpd to the latest version, 2.3.1.

Fixes https://github.com/opnsense/ports/issues/160 and https://github.com/opnsense/plugins/issues/3163

Has been tested locally as working OK on opnsense 22.7.6

### CODE REVIEW

File: net/miniupnpd/Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ PLIST_FILES=	etc/miniupnpd.conf.sample \
 		man/man8/miniupnpd.8.gz \
 		sbin/miniupnpd
 
-OPTIONS_DEFINE=			CHECK_PORTINUSE IPV6 LEASEFILE UPNP_IGDV2 \
+OPTIONS_DEFINE=			PF_FILTER_RULES CHECK_PORTINUSE IPV6 LEASEFILE UPNP_IGDV2 \
```
Comment (fichtner) [inline net/miniupnpd/Makefile:28]: doesn't look like this serves a functional change and we set "PF" through tools.git:

config/22.7/make.conf:net_miniupnpd_SET=                CHECK_PORTINUSE PF

**Comment by fichtner** [comment]
ideally this should be funnelled through https://bugs.freebsd.org but I can help with that

**Comment by fichtner** [comment]
@Tawmu I shifted this to a development package, instructions to come in plugins PR, see d0b581e4fb9

**Comment by Tawmu** [comment]
@fichtner just to be clear - the work in https://github.com/opnsense/plugins/pull/3096 is completely seperate to the issue of fixing game console issues, that is mainly about UPNP behind L3 switching.

To fix game consoles/lobbies this miniupnpd update combined with https://github.com/opnsense/plugins/pull/3165 is all that is needed.

**Comment by fichtner** [comment]
@Tawmu ok, I've changed it for 22.7.10 that the development version os-unpnp-devel will have the correct dependency.

**Comment by effex12** [comment]
> @fichtner just to be clear - the work in [opnsense/plugins#3096](https://github.com/opnsense/plugins/pull/3096) is completely seperate to the issue of fixing game console issues, it's mainly about UPNP behind L3 switching.
> 
> To fix game consoles/lobbies this miniupnpd package update combined with [opnsense/plugins#3165](https://github.com/opnsense/plugins/pull/3165) is all that is needed.

Hi @Tawmu , how can I apply this now? Do I need to wait until the next release?

**Comment by Tawmu** [comment]
> > @fichtner just to be clear - the work in [opnsense/plugins#3096](https://github.com/opnsense/plugins/pull/3096) is completely seperate to the issue of fixing game console issues, it's mainly about UPNP behind L3 switching.
> > 
> > To fix game consoles/lobbies this miniupnpd package update combined with [opnsense/plugins#3165](https://github.com/opnsense/plugins/pull/3165) is all that is needed.
> 
> Hi @Tawmu , how can I apply this now? Do I need to wait until the next release?

I believe you can install the os-upnp-devel package in the latest version of opnsense

**Comment by effex12** [comment]
> > > @fichtner just to be clear - the work in [opnsense/plugins#3096](https://github.com/opnsense/plugins/pull/3096) is completely seperate to the issue of fixing game console issues, it's mainly about UPNP behind L3 switching.
> > > To fix game consoles/lobbies this miniupnpd package update combined with [opnsense/plugins#3165](https://github.com/opnsense/plugins/pull/3165) is all that is needed.
> > 
> > 
> > Hi @Tawmu , how can I apply this now? Do I need to wait until the next release?
> 
> I believe you can install the os-upnp-devel package in the latest version of opnsense

Apologies for the dumb question, but I'm not able to see -devel package in the latest version of opnsense GUI.

Is this something I'd need to SSH into the box and do pkg install to get?

Also -- will I need to uninstall the existing os-upnp package? 

I've been experiencing an issue with UPnP that ultimately led me here and I believe this is the fix. (Adding the NAT anchors to the rules)

Thank you!

**Pull request closed without merging**